### PR TITLE
MRC-114 Error 500 Page Content

### PR DESCRIPTION
**Pull request by adamjw3**

### CODE REVIEW

File: conf/messages.en [hunk 1]
```diff
@@ -229,4 +229,7 @@ distributionsIncluded.emptyValue = None
 session.expired.title = Session expired
 session.expired.heading = Your session has ended due to inactivity
 session.expired.body = You have not entered any details for {0} minutes, so we''ve cleared your information.
-session.expired.btn.restart = Start again
\ No newline at end of file
+session.expired.btn.restart = Start again
+
+page.error.500.heading = Sorry, there is a problem with the service
+page.error.500.message = Try again later.<br/><br/>We have not saved any of your answers. When the service is available, you will have to start again.
```
Comment (girishkamat) [inline conf/messages.en:235]: Is there a way to do this without adding HTML into text?

**Review by PavelVjalicin — CHANGES_REQUESTED** [review]
We probably need to add a test for internal server error.

**Review by PavelVjalicin — APPROVED** [review]

File: app/views/ErrorTemplate.scala.html [hunk 2]
```diff
@@ -23,5 +23,5 @@
 @layout(pageTitle = titleNoForm(pageTitle)) {
     <h1 class="govuk-heading-xl">@messages(heading)</h1>
 
-    <p class="govuk-body">@messages(message)</p>
+    <p class="govuk-body">@Html(messages(message))</p>
```
Comment (girishkamat) [inline app/views/ErrorTemplate.scala.html:26]: Does this need to removed i.e @Html

File: app/handlers/ErrorHandler.scala [hunk 3]
```diff
@@ -17,20 +17,25 @@
...
+  errorInternal: ErrorInternalView
 ) extends FrontendErrorHandler with I18nSupport {
 
   override def standardErrorTemplate(pageTitle: String, heading: String, message: String)(implicit
     rh: Request[_]
   ): Html =
     view(pageTitle, heading, message)
+
+  override def internalServerErrorTemplate(implicit request: Request[_]): Html =
+    errorInternal()
```
Comment (girishkamat) [inline app/handlers/ErrorHandler.scala:40]: Can you remind me when is this error triggerred?

File: app/controllers/ErrorInternalController.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+package controllers
+
+import controllers.actions._
+import javax.inject.Inject
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.ErrorInternalView
+
+class ErrorInternalController @Inject()(
```
Comment (girishkamat) [inline app/controllers/ErrorInternalController.scala:26]: Why do we need this controller?

File: app/handlers/ErrorHandler.scala (hunk 3 above)
Comment (adamjw3) [reply app/handlers/ErrorHandler.scala:40]: This is server error 500, so to test it i just stopped the backend server

File: app/controllers/ErrorInternalController.scala (hunk 4 above)
Comment (adamjw3) [reply app/controllers/ErrorInternalController.scala]: Removed

File: app/views/ErrorInternalView.scala.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
+@*
```
Comment (girishkamat) [inline app/views/ErrorInternalView.scala.html:1]: Can we call this InternalServerErrorTemplate instead, as it seems to be specifically used for 500 internal server errors

**Comment by girishkamat** [comment]
Looks good, except for for a minor comment regarding name of view file

**Review by girishkamat — APPROVED** [review]

**Pull request merged**